app/src/main/java/com/android/streetworkapp/ui/profile/Settings.kt

```diff
@@ -87,3 +96,29 @@ fun SettingsContent(
         // navigationActions.navigateTo(Screen.AUTH)
       })
 }
+
+/**
+ * Logs out the user by signing out from Google and the app.
+ *
+ * @param authService the Google authentication service
+ * @param userViewModel the user viewmodel
+ * @param preferencesViewModel the preferences viewmodel
+ */
+fun logout(
+    authService: GoogleAuthService,
+    userViewModel: UserViewModel,
+    preferencesViewModel: PreferencesViewModel
+) {
+  // Sign out from Google Service (disable auto-login)
+  authService.signOut()
+
+  // Clear user viewmodel data
+  userViewModel.setUser(null)
+  userViewModel.setCurrentUser(null)
+
+  // Clear preferences parameters
+  preferencesViewModel.setLoginState(false)
+  preferencesViewModel.setUid("")
+  preferencesViewModel.setName("")
+  preferencesViewModel.setScore(0)
+}
```

Comment:
Improvement: The data is removed inside the PreferencesViewModel, but when I try to re-login using the Google button, I am automatically logged in with my previous account. 
After some research, I found that the Google access token needs to be revoked to fully log out the user. 
This can be done using the following method:
```kotlin
authService.revokeAccess()
```
I may have done something wrong, but it's strange that I can't choose which google account to log in with. What do you think ?

Reply:
Good catch and thanks for the video, it helped me understand the problem. In fact, the signout function of the `GoogleAuthService` class didn't take care of properly resetting the client, so it wasn't properly disconnected and therefore it wasn't possible to choose a Google account to log in. I've made the changes.

I'll let you test if it works on your side. Let me know if you have any other questions or suggestions.